src/fileio/datafile.cxx

```diff
@@ -204,7 +208,21 @@ bool Datafile::openw(const std::string& filename_) {
       throw BoutException("Failed to add int variable {:s} to Datafile", var.name);
     }
   }
+
+  // Add vectors of integers
+  for(const auto& var : int_vec_arr) {
+    if (!file->addVarIntVec(var.name, var.save_repeat, var.ptr->size())) {
+      throw BoutException("Failed to add int vector variable {:s} to Datafile", var.name);
+    }
```

Comment:
I've added some checks that the size of the `std::vector`s haven't been changed when adding/reading/writing. Possibly we could make this a warning and use the smaller size, but that risks losing information, so I think an error forcing the user to fix the input is safer.

Reply:
Yes, explicit error is good